src/components/search/InputWithSuggester.tsx

```diff
@@ -178,37 +191,29 @@ const InputWithSuggester: FC<InputWithSuggesterProps> = ({
     };
   }, [debounceRefreshOptions]);
 
-  const getGroup = useCallback(
+  const getGroup: (option: string) => string | undefined = useCallback(
     (option: string) => {
       return options.find((o) => o.text.toLowerCase() === option.toLowerCase())
         ?.group;
     },
     [options]
   );
 
-  const onChange = useCallback(
-    (_: any, newValue: string | null) => {
-      if (newValue !== null) {
-        // String quote with double quote to indicate user want the whole phase during search.
-        // fire event here before useState update, need to call function in this case
-        if (getGroup(newValue) === OptionGroup.CATEGORY) {
-          addCategory(newValue);
-          dispatch(updateSearchText(""));
-        }
-      }
-    },
-    [addCategory, dispatch, getGroup]
-  );
-
   const onInputChange = useCallback(
-    (_: any, newInputValue: string) => {
+    async (_: any, newInputValue: string) => {
       // If user type anything, then it is not a title search anymore
       dispatch(updateSearchText(newInputValue));
       if (newInputValue?.length > 1) {
-        debounceRefreshOptions()?.then();
+        await debounceRefreshOptions(); // wait for the debounced refresh to complete
+        // TODO: getGroup has delay, needs to instantly return group value or await until getGroup returns,
+        //  otherwise, commonKey will be falsely ignored in the store (e.g. you type 'salinity' and quickly select it from the list, but it is not recognised as common key)
```

Comment:
@NekoLyn feel free to address this TODO the way you think the most optimal 

@utas-raymondng FYI, keep you in the loop